ecg_data.py

```diff
@@ -1,12 +1,100 @@
-class hrm:
+import numpy
+import pandas
+
+
+class ecg_data:
     'This is a class'
-    def __init__(self, filename, avgT1,avgT2,bradyT,bradyThresh,tachyT,tachyThresh):
-        self.avgT1 = 0.0
-        self.avgT2 = 10.0
 
-    #def instantHr(self):
+    def __init__(
+            self,
+            filename,
+            begin_time=0,
+            end_time=10,
+            bradyT=5,
+            bradyThresh=60,
+            tachyT=5,
+            tachyThresh=100):
+
+        # put peak detection here
+        data = pandas.read_csv(
+            filename, converters={
+                "times": float, "voltages": float})
+        voltages = data.voltages.values
+        finalTimes = [0]
+        avgvoltage = numpy.average(voltages)
+
+        threshvoltage = avgvoltage * 2
+
+        peaks = numpy.where(voltages >= threshvoltage)
+        peaks1 = peaks[0]
+        peaks2 = peaks1.tolist()
+        boxcar = []
+        for i in range(1, len(peaks2) - 1):
+            if (voltages[peaks2[i]] >= voltages[peaks2[i - 1]]) and \
+                    (voltages[peaks2[i]] >= voltages[peaks2[i + 1]]):
+                recentval = finalTimes[len(finalTimes) - 1]
+                finalTimes.append(peaks2[i])
+                if (peaks2[i] - recentval <= 50):
+                    finalTimes.pop()
+        finalTimes.pop(0)
+
+        self.times = finalTimes
+        self.inst = self.instantHr(begin_time)
+        self.avg = self.averageHR(begin_time, end_time)
+        self.ano = self.anomalyHr(bradyT, bradyThresh, tachyT, tachyThresh)
+
+    def instantHr(self, target_time = 0):
+        for x in range(0, len(self.time)):
+            if self.time[x] > target_time:
+                instant_dt = self.time[x + 1] - self.time[x]
+                break
+
+        return (60 / instant_dt) * 100
+
+    def averageHr(self, begin_time = 0, end_time = 10):
+        for j in range(1, len(self.time)):
+            if (self.time[j - 1] / 1000 == begin_time):
+                begin = j - 1
+            elif (self.time[j - 1] / 1000 < begin_time) and \
+                    (self.time[j] / 1000 > begin_time):
+                begin = j
+            if (self.time[j - 1] / 1000 == end_time):
+                end = j - 1
+            elif (self.time[j - 1] / 1000 < end_time) and \
+                    (self.time[j] / 1000 > end_time):
+                end = 1
+        time_count = 0
+
+        for k in range(begin + 1, end + 1):
+            time_count = time_count + (self.time[k] - self.time[k - 1]) / 1000
+
+        div = end - begin
 
-    #def averageHr(self):
+        time_avg = time_count / div
+        return 60 / time_avg
 
-    #def anomalyHr(self):
+    def anomalyHr(self, bradyT = 5, bradyThresh = 60, tachyT = 5, tachyThresh = 100):
+        dying_slow = 0
+        dying_fast = 0
+        bradyTimes = []
+        tachyTimes = []
+        counter = 0
+        for l in range(1, len(self.time)):
```

Comment:
oh my... definitely would try to refactor this for readability

Reply:
@syiyuli